sim/firesim-lib/src/main/scala/bridges/TerminationBridge.scala

```diff
@@ -0,0 +1,133 @@
+
+class TerminationBridgeTargetIO(params: TerminationBridgeParams) extends Bundle {
+  val clock = Input(Clock())
+  val errCode = Input(UInt((log2Ceil((params.conditionInfo).size)).W))
+  val valid = Input(Bool()) 
+}
+
+class TerminationBridgeHostIO(params: TerminationBridgeParams)
+                              (private val targetIO : TerminationBridgeTargetIO = new TerminationBridgeTargetIO(params)) 
+                               extends Bundle with ChannelizedHostPortIO{
+  def targetClockRef = targetIO.clock 
+  //It is just inputs from the target to indicate test completion
+  //The HostPort annotation would not work with aggregates so creating individual channels.
+  val errCodeH = InputChannel(targetIO.errCode)
+  val validH = InputChannel(targetIO.valid)
+}
+
+class TerminationBridge(params: TerminationBridgeParams) extends BlackBox
+    with Bridge[TerminationBridgeHostIO, TerminationBridgeModule] {
+  val io = IO(new TerminationBridgeTargetIO(params))
+  val bridgeIO = new TerminationBridgeHostIO(params)(io)
+
+  val constructorArg = Some(params)
+
+  generateAnnotations()
+}
+
+object TerminationBridge {
+
+  private def annotateTerminationBridge(clock: Clock, reset: Reset, conditionBools: Seq[Bool], params: TerminationBridgeParams): TerminationBridge = {
+    val finish = conditionBools.reduce(_||_)
+    val errMessageID = PriorityEncoder(conditionBools)
+    //val gReset = WireInit(false.B)
+    //GlobalResetCondition.sink(gReset)
+    require(params.conditionInfo.size==conditionBools.size)
+    val terminationBridgeTarget = Module(new TerminationBridge(params))
+    terminationBridgeTarget.io.errCode := errMessageID
+    terminationBridgeTarget.io.valid := finish && !reset.asBool
+    terminationBridgeTarget.io.clock := clock 
+    terminationBridgeTarget  
+  }
+
+/** Instatiates the target side of the Bridge, selects one of 
+ *  the available finish conditions and passes on the corresponding 
+ *  error message.
+ *  @param clock: Clock to the bridge which it must run on.
+ *  @param reset: local reset for the bridge.
+ *  @param conditionBools: Seq of finished conditions indicated by running programs
+ *  @param params: Possible list of message info to be returned by simulator  
+ */
+
+  def apply(clock: Clock, reset: Reset, conditionBools: Seq[Bool], params: TerminationBridgeParams): TerminationBridge = {
+    annotateTerminationBridge(clock, reset, conditionBools, params)
+  }
+
+/** Simpler way to instantiate target side of the bridge by using implicit clock and reset
+ */
+  def apply(conditionBools: Seq[Bool], params: TerminationBridgeParams): TerminationBridge = {
+    annotateTerminationBridge(Module.clock, Module.reset, conditionBools, params)
+  }
+
+}
+
+class TerminationBridgeModule(params: TerminationBridgeParams)(implicit p: Parameters) extends BridgeModule[TerminationBridgeHostIO]()(p) {
+  lazy val module = new BridgeModuleImp(this) {
+    // This creates the interfaces for all of the host-side transport
```

Comment:
This comment is no longer accurate. I'd just say it instantiates the control bus.

Reply:
You can just delete it if you're unsure.